fluster/image/raw_buffer.py

```diff
@@ -0,0 +1,361 @@
+    ) -> int:
+        """Comparison of two buffers of type YUV420"""
+        source_buffer: bytes = self.buffer
+        target_buffer: bytes = target.buffer
+        size_chroma: int = self.frame_size // 4
+        start_u: int = self.frame_size
+        start_v: int = self.frame_size + size_chroma
+        total_errors = 0
+
+        def check_deviation(source_pixel: int, target_pixel: int) -> int:
+            deviation = int(abs(source_pixel - target_pixel))
+            if deviation > self._YUV_TOLERANCE:
+                if i not in self._pixels_to_ignore:
+                    deviations[deviation] = (
+                        1 if deviation not in deviations else deviations[deviation] + 1
+                    )
+                    return 1
+            return 0
+
+        # First compare the y buffer
+        for i in range(self.frame_size):
+            total_errors += check_deviation(source_buffer[i], target_buffer[i])
+            if i < size_chroma:
+                total_errors += check_deviation(
+                    source_buffer[i + start_u], target_buffer[i + start_u]
+                )
+                total_errors += check_deviation(
+                    source_buffer[i + start_v], target_buffer[i + start_v]
+                )
+
+        return total_errors
+
+    def _rgba_comparison(self, target: "RawBuffer", deviations: Dict[int, int]) -> int:
+        """Comparison of two buffers of type RGBA"""
+        if self._create_diff_buffer:
+            self._buffer_diff = bytearray(self._buffer)
+        total_errors = 0
+
+        for i in range((self.frame_size * 4)):
+            deviation = abs(self._buffer[i] - target.buffer[i])
+            if deviation:
+                total_errors += 1
+                if self._create_diff_buffer and i % 4 == 0:
+                    self._buffer_diff[i] = self._color_diff[0]
+                    self._buffer_diff[i + 1] = self._color_diff[1]
+                    self._buffer_diff[i + 2] = self._color_diff[2]
+                deviations[deviation] = (
+                    1 if deviation not in deviations else deviations[deviation] + 1
+                )
+
+        return total_errors
+
+    @staticmethod
+    def clamp(value: int) -> int:
+        """Clip between 0 and 255"""
+        return 255 if (value > 255) else (0 if (value < 0) else value)
+
+    def save(self, filename: str) -> None:
+        """Save the content of the buffer to file"""
+        with open(filename, "wb") as file:
+            file.write(self._buffer)
+
+    def save_diff(self, filename: str) -> None:
+        """Save the content of the buffer to file"""
+        with open(filename, "wb") as file:
+            file.write(self._buffer_diff)
+
+    def save_as_python(self, filename: str) -> None:
+        """Save the content of the buffer to file"""
+        with open(filename, "w") as file:
+            counter = 0
+            sentence = "image_content = ["
```

Comment:
It is only a help to see how the image buffer is, I have used it in the tests of the OCP-2303, but it is true that now I do not need it. It can be removed